test/test_judiVStack.jl

```diff
@@ -0,0 +1,80 @@
+# Unit test for judiVStack
+# Ziyi Yin (ziyi.yin@gatech.edu)
+# Nov 2020
+
+using JUDI.TimeModeling, LinearAlgebra, PyPlot, IterativeSolvers, JOLI, Test, Printf
```

Comment:
This is a modeling test, this is not testing the unit your adding wich is `I * w` that does not require any of this, Check how the other datatypes tests are done. This file need to be removed.

Reply:
This tests joCoreBlock*judiVStack and the adjoint. I think it's necessary (since before, jo_check_type didn't pass)

Reply:
It needs a test but this is not a good one, this is overly complicated for the unit you are testing

Reply:
All you should need os

```
c = [w1; jw2]
A = [joDirac; joSomething]
@test A*c = [w1, joSomething*w2]
```

and that's the only thing you should have to test in `test_composite`. Every test should always be as minimal as possible. And for example every sing test in `GROUP=JUDI` does not require and therefore does not install devito

Reply:
I am still a bit confused at here. To my understanding, this pull request mainly deals with the problem where joDirac type of operators output array instead of judiWeights, which joCoreBlock operator does not like. But, for any other type of joLinearFunction, when we do multiplication (either with judiVector or Weights or even PhysicalParameters), we have to output array in default because we don't have any assumption on the output. Therefore, I don't have any idea on how to make this test now -- seems the only meaningful test is to set `A = [lambda1*joDirac; lambda2*joDirac]`and test if A*c makes sense. Any other joSomething would be expected to fail the test. Then at the end of the day, we are not really testing joCoreBlock, we are just simply testing if joDirac outputs judiWeights instead of array.

Also, is it reasonable to overload multiplications of joDirac for judiVector as well? I am not sure